packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/getLinkedObject.ts

```diff
@@ -47,7 +47,11 @@ export function getLinkedObject<T extends OntologyObject>(
           select: [],
         },
       );
-      return convertWireToOsdkObject(client, linkTypeApiName, object) as T;
+      return convertWireToOsdkObject(
+        client,
+        linkTypeApiName,
+        object,
+      ) as unknown as T;
```

Comment:
This is a side effect of having `OntologyObject` be parametrized with parameter value, I'm tracking cleaning these up in a separate branch